### ISSUE TITLE
Bad results

### ISSUE THREAD

**Issue by snarb** [minar09/cp-vton-plus#26]
Thank for the work. 
I have run  the  modules according to the read me, but most of  results after the TOM stage ('try-on' generated folder) contains significant artifacts. Just attaching here first 5 generated photos. Some are ok, but I would say that most are bad.  Don't understand  is this a some bug, problem with my actions or current limitation of the approach.
![000001_0](https://user-images.githubusercontent.com/4518007/93223246-8522ad80-f778-11ea-8e52-34846a582dd7.jpg)
![000010_0](https://user-images.githubusercontent.com/4518007/93223252-85bb4400-f778-11ea-81c5-4cc29723f90a.jpg)
![000020_0](https://user-images.githubusercontent.com/4518007/93223253-8653da80-f778-11ea-908d-ac31312679fb.jpg)
![000028_0](https://user-images.githubusercontent.com/4518007/93223254-8653da80-f778-11ea-8374-f22bfa4c5d11.jpg)
![000038_0](https://user-images.githubusercontent.com/4518007/93223256-8653da80-f778-11ea-9297-4569281cb5c9.jpg)

**Comment by snarb** [comment]
There some weird samples in dataset. This is first close mask in the dataset. It contains two closes 
![000001_1](https://user-images.githubusercontent.com/4518007/93228006-b05bcb80-f77d-11ea-97f5-77fd37400a15.jpg)
Images in image-mask folder often looks like combination of two photos.. 
![000020_0](https://user-images.githubusercontent.com/4518007/93228275-fa44b180-f77d-11ea-800f-1589354ce0db.png)
will check

**Comment by thaithanhtuan** [comment]
Please refer to other issues for more information:
https://github.com/minar09/cp-vton-plus/issues/9
And this:
https://github.com/minar09/cp-vton-plus/issues/8
Somebody has the same problem and comes from a different OS return "listDir" function.
-------------
![image](https://user-images.githubusercontent.com/19486377/93280143-dd52c180-f803-11ea-892c-8c56b8de2818.png)
Yes, In VITON dataset, there are some weird cases, you can simply remove them from train_pair.txt

**Comment by vasujoshi111** [comment]
Hi, I was trying on custom images for prediction using pretrained models provided in repo.
I was getting a bad results.
Only for getting the keypoints, I used detectron2 and added one extra list like [[0,0,0]](As I was getting 17 keypoints).
Changed the batch size to 1 also, and tried for gray scale images also.
Please help me out in getting a better results.
Keypoints json:
{"people": [{"face_keypoints": [], "hand_left_keypoints": [], "hand_right_keypoints": [], "pose_keypoints": [115.9163818359375, 41.370147705078125, 0.9661189317703247, 126.4640884399414, 31.777069091796875, 4.307704448699951, 106.80697631835938, 31.29741668701172, 2.1659820079803467, 140.367919921875, 36.5736083984375, 0.78687983751297, 95.54009246826172, 35.8541259765625, 0.6354686617851257, 159.54559326171875, 95.57103729248047, 0.15054160356521606, 74.44464874267578, 90.77449035644531, 0.27401798963546753, 163.86058044433594, 168.47842407226562, 0.11921537667512894, 69.65023040771484, 150.01174926757812, 0.04564519599080086, 140.367919921875, 234.67066955566406, 0.21882663667201996, 98.41674041748047, 194.37973022460938, 0.0308991651982069, 126.4640884399414, 215.00485229492188, 0.07232855260372162, 71.3282699584961, 212.84640502929688, 0.0809921994805336, 114.7177734375, 251.93820190429688, 0.0634264126420021, 67.01329803466797, 251.93820190429688, 0.05943075194954872, 133.41600036621094, 251.93820190429688, 0.10058610886335373, 48.554779052734375, 251.93820190429688, 0.028173452243208885, 0, 0, 0]}], "version": 1.0}

Result
![000010_0](https://user-images.githubusercontent.com/71640167/93757985-46cc3900-fc25-11ea-82df-49d4cbf54362.jpg)
Cloth
![000028_1](https://user-images.githubusercontent.com/71640167/93757990-492e9300-fc25-11ea-87ec-595f11720169.jpg)
Image
![000010_0 (1)](https://user-images.githubusercontent.com/71640167/93757991-49c72980-fc25-11ea-8579-033ddc6ac9ef.jpg)
Image-parse
![000010_0](https://user-images.githubusercontent.com/71640167/93757995-4a5fc000-fc25-11ea-9edd-2003c9559735.png)

**Comment by minar09** [comment]
Hi @vasujoshi111 , is the detectron2 joints ordering are same as openose? If not, you need to generate poses for your images with openpose coco-18 model.

**Comment by vasujoshi111** [comment]
Thank you very much @minar09 ,  I need to know about this. I will try to get the difference.

**Comment by vasujoshi111** [comment]
Hi @minar09 , Even with viton data with pretrained model, also I am getting bad results. PFA result images.
![000395_0](https://user-images.githubusercontent.com/71640167/94007868-b37b3b00-fdbf-11ea-80cf-b65a5d500f02.jpg)
![000477_0](https://user-images.githubusercontent.com/71640167/94007872-b413d180-fdbf-11ea-968e-3f484bb2534f.jpg)
![000048_0](https://user-images.githubusercontent.com/71640167/94007856-af4f1d80-fdbf-11ea-8d7b-cb4fac65217d.jpg)
![000183_0](https://user-images.githubusercontent.com/71640167/94007864-b118e100-fdbf-11ea-9418-72afa211329b.jpg)
![000248_0](https://user-images.githubusercontent.com/71640167/94007866-b24a0e00-fdbf-11ea-905c-a5bdd8bb5094.jpg)

**Comment by minar09** [comment]
Hi @vasujoshi111 , your results are obviously not as expected. Some people previously had sorting errors, but this one looks different. Please debug a little to see whether all the inputs are coming as expected. I think somehow the face-hair input is missing from your TOM network. Hope you can solve this. Good luck.

**Comment by vasujoshi111** [comment]
Thanks @minar09 . I spotted the error. The error is in the parsed image. Please share the link or code to get the exact parse image as in vton dataset.
I have used https://github.com/RohanBhandari/LIP_JPPNet to get the parse image. But that is not same as the one in vton dataset parse image.

**Comment by minar09** [comment]
I think VITON dataset segmentation is generated with this: https://github.com/Engineering-Course/LIP_SSL
Please check the original paper for more details.

**Comment by vasujoshi111** [comment]
Thank you @minar09 . while debugging the code, in dataset_neck_skin_correction.py line no. 132 Image.open(seg_pth), our generated image which is exactly same as cp-vton data image having 3 channels. but cp-vton parse-image is having shape 256*192 in "P" mode. I converted our generated image to P mode and run. Result is not good. Changed to "L" mode then I got upper neck part. Every image is same till neck_mask in dataset_neck_skin_correction.py(line no. 181) but after adding the segmentation to neck mask I got different image as shown below.
![result](https://user-images.githubusercontent.com/71640167/94530185-5c121a80-0258-11eb-972a-ef27af9c8e54.PNG)
After passing this image to decode_labels I am getting this one.
![decode](https://user-images.githubusercontent.com/71640167/94530281-819f2400-0258-11eb-8a79-b7d175c81317.PNG)
How to fix this one?

**Comment by minar09** [comment]
@vasujoshi111 , you don't need to run `dataset_neck_skin_correction.py` anymore. We already provide fully processed dataset for downloading in the readme file. You can directly move to training/testing.

**Comment by vasujoshi111** [comment]
@minar09 , I am running each line of code in dataset_neck_skin_correction.py in order to know how can generate for custom images. In custom images prediction I got weird results. That's why I am trying to get how they have processed the image-parse and all. I have taken one of the image from vton data and trying to get the same images as in vton dataset and get your results.
